synapse/axon.py

```diff
@@ -83,7 +85,34 @@ async def get(self, sha256):
         resp = await self.cell.has(s_common.uhex(sha256))
         return self.sendRestRetn(resp)
 
-class AxonHttpDownloadV1(s_httpapi.Handler):
+reqValidAxonDel = s_config.getJsValidator({
+    'type': 'object',
+    'properties': {
+        'sha256s': {
+            'type': 'array',
+            'items': {'type': 'string', 'pattern': '(?i)^[0-9a-f]{64}$'}
+        },
+    },
+    'additionalProperties': False,
+    'required': ['sha256s'],
+})
+
+class AxonHttpDelV1(s_httpapi.Handler):
+
+    async def post(self):
+
+        if not await self.reqAuthAllowed(('axon', 'del')):
+            return
+
+        body = self.getJsonBody(validator=reqValidAxonDel)
+        if body is None:
+            return
+
+        hashes = [s_common.uhex(s) for s in body.get('sha256s')]
+        resp = await self.cell.dels(hashes)
+        return self.sendRestRetn(resp)
```

Comment:
can we zip this together to make a dictionary we return? hash -> bool?

Reply:
Zipped as array of tuples since there's no guarantee the input is unique, e.g. if the list contains a sha256 twice the first will be True and second False, so I think this way keeps the input context in place

Reply:
There's a data compression issue here... but i guess it's fine :D